gnutls: 3.8.6 -> 3.8.9

MisileLab opened the pull request:
https://gitlab.com/gnutls/gnutls/-/blob/3.8.9/NEWS
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MisileLab committed:
gnutls: 3.8.6 -> 3.8.9

vcunat commented:
I've been following their atom feed, but last one in there is 3.8.4 😠

MisileLab commented:
> I've been following their atom feed, but the last version in there is 3.8.4 😠 

honestly, I found many upgrades (including this) from archlinux and scoop (windows)

MisileLab commented:
patch failed
```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/x750diyxk4x1x6nl0ch5qh71cj9pmdji-gnutls-3.8.9.tar.xz
source root is gnutls-3.8.9
setting SOURCE_DATE_EPOCH to timestamp 1738990782 of file "gnutls-3.8.9/doc/gnutls.info"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/fb56szw1cn21s5py4av2vnc7rjxbna7w-nix-ssl-cert-file.patch
patching file lib/system/certs.c
Hunk #1 succeeded at 381 with fuzz 2 (offset -23 lines).
applying patch /nix/store/bd3xgwcxc5w5km6wka7wx8mfq581f9ng-revert-dlopen-compression.patch
patching file lib/Makefile.am
Hunk #1 FAILED at 156.
1 out of 1 hunk FAILED -- saving rejects to file lib/Makefile.am.rej
patching file lib/compress.c
Hunk #2 FAILED at 20.
Hunk #3 FAILED at 41.
Hunk #4 succeeded at 145 (offset -107 lines).
Hunk #5 FAILED at 314.
Hunk #6 FAILED at 341.
Hunk #7 FAILED at 352.
Hunk #8 FAILED at 364.
Hunk #9 FAILED at 394.
Hunk #10 FAILED at 405.
Hunk #11 FAILED at 416.
9 out of 11 hunks FAILED -- saving rejects to file lib/compress.c.rej
patching file lib/compress.h
patching file lib/ext/compress_certificate.c
patching file lib/global.c
Hunk #2 succeeded at 416 (offset 1 line).
patching file tests/tls13/compress-cert-cli.c
Hunk #1 FAILED at 23.
1 out of 2 hunks FAILED -- saving rejects to file tests/tls13/compress-cert-cli.c.rej
```

MisileLab commented:
I updated patch

MisileLab commented:
@vcunat revert dlopen failed, maybe we need to make it patch file instead of fetchpatch

vcunat commented:
I have a vague feeling that they later change the behavior to something sensible, so I'd expect no patching of this.  But I'll have to look.

vcunat committed:
gnutls: adapt linking after 3.8.8

Upstream:
** build: More flexible library linking options for compression libraries, TPM, and liboqs support
   The configure options, --with-zstd, --with-brotli, --with-zlib,
   --with-tpm2, and --with-liboqs now take 4 states:
   yes/link/dlopen/no, to specify how the libraries are linked or
   loaded.

I haven't figure out how to easily avoid ENABLE_DLOPEN from configure,
but we don't have the other libraries so far, so it's simple anyway.

vcunat reviewed (approved):
I went through NEWS, in particular looked int the linking topic, built `gnutls.tests` atop nixpkgs master (those that [evaluated](https://github.com/NixOS/nixpkgs/pull/215868#issuecomment-2693859338)).
